packager/VisibilityChecker.cc

```diff
@@ -18,9 +18,21 @@ namespace sorbet::packager {
 
 namespace {
 
+static core::SymbolRef getEnumClassForEnumValue(const core::GlobalState &gs, core::SymbolRef sym) {
+    if (sym.isStaticField(gs) && sym.owner(gs).isClassOrModule()) {
+        auto owner = sym.owner(gs);
+        if (owner.asClassOrModuleRef().data(gs)->superClass() == core::Symbols::T_Enum()) {
```

Comment:
```suggestion
        // There's a hidden class like `MyEnum::X$1` between `MyEnum::X` and `T::Enum` in the ancestor chain.
        if (owner.asClassOrModuleRef().data(gs)->superClass() == core::Symbols::T_Enum()) {
```

Reply:
Sorry, don't quite understand this comment.

Reply:
I just wanted to leave a note why you need the `->superClass()` here. naively, you might expect `enum.owner(gs)` to be `T_Enum` but that's not how it works, because of the rewrites we do in `TEnum.cc` to support exhaustiveness on enums. feel free to reword this comment however you want to capture why we need the `->superClass` here.